pkg/getter/httpgetter_test.go

```diff
@@ -37,12 +41,47 @@ func TestHTTPGetter(t *testing.T) {
 	cd := "../../testdata"
 	join := filepath.Join
 	ca, pub, priv := join(cd, "ca.pem"), join(cd, "crt.pem"), join(cd, "key.pem")
-	g, err = newHTTPGetter("http://example.com/", pub, priv, ca)
+	g, err = newHTTPGetter(Options{
+		URL:      "http://example.com",
+		CertFile: pub,
+		KeyFile:  priv,
+		CAFile:   ca,
+	})
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if _, ok := g.(*HTTPGetter); !ok {
+	hg, ok := g.(*HTTPGetter)
+	if !ok {
 		t.Fatal("Expected newHTTPGetter to produce an httpGetter")
 	}
+
+	transport, ok := hg.client.Transport.(*http.Transport)
+	if !ok {
+		t.Errorf("Expected newHTTPGetter to set up an HTTP transport")
+	}
+
+	test.AssertGoldenString(t, transport.TLSClientConfig.ServerName, "output/httpgetter-servername.txt")
+
+	// Test other options
+	hg, err = NewHTTPGetter(Options{
```

Comment:
Can you use different values for the uname, passwd and ua to be sure they are set correctly and therefore the check below can prove it?

Reply:
sure.